client/app/actions/ManageAvailabilityActions.js

```diff
@@ -17,31 +18,74 @@ const changeVisibleMonth = (day) => ({
   payload: day,
 });
 
-export const dataLoaded = (slots) => ({
+export const dataLoaded = (slots, loadedMonths) => ({
   type: actionTypes.DATA_LOADED,
-  payload: slots,
+  payload: slots.merge({ loadedMonths }),
 });
 
+/**
+   Number of extra months to preload.
+   0 mean no preloading.
+ */
+const PRELOAD_MONTHS = 2;
+
+/**
+   Given a `date` and number of months to preload, return a List of
+   moment dates representing the start of the month that should be
+   loaded.
+
+   @param {moment} date - Date representing the month
+   @params {number} preloadMonths - Number of months to preload
+   @return {List[moment]} List of months that should be loaded
+ */
+const loadRange = (date, preloadMonths) => {
+  const startOfMonth = date.clone().startOf('month');
+  const endOfMonthExclusive = startOfMonth.clone().add(1, 'month');
+
+  const start = startOfMonth.clone().subtract(preloadMonths, 'months');
+  const end = endOfMonthExclusive.clone().add(preloadMonths, 'months');
+
+  return expandRange(start, end, 'months');
+};
+
+/**
+   Given a list of `months` and a Set of already `loadedMonths`,
+   return an object { start, end } where already loaded months is removed.
+ */
```

Comment:
This comment could be cleared. The code removes already loaded months from the start and end of the range, but there can be some loaded months in the middle.